spec/unit/actions/synchronize_etls_in_segment_spec.rb

```diff
@@ -30,20 +47,97 @@
 
   before do
     allow(gdc_gd_client).to receive(:domain) { domain }
-    allow(gdc_gd_client).to receive(:projects) { project }
+    allow(gdc_gd_client).to receive(:projects).with('from project') { project }
+    allow(gdc_gd_client).to receive(:projects).with(target_project_id) { project }
     allow(project).to receive(:schedules) { [] }
     allow(data_product).to receive(:segments) { [segment] }
     allow(segment).to receive(:synchronize_processes) { result }
     allow(project).to receive(:set_metadata)
     allow(segment).to receive(:segment_id) { 'some_segment_ids' }
+    allow(project).to receive(:processes) { [target_process] }
+    allow(process).to receive(:process_id) { process_id }
+    allow(process).to receive(:name) { process_name }
+    allow(gdc_gd_client).to receive(:projects)
+      .with(target_project_id) { target_project }
+    allow(target_project).to receive(:processes) { [target_process, custom_target_process] }
+    allow(target_project).to receive(:set_metadata)
+    allow(target_project).to receive(:schedules) { [custom_target_schedule, target_schedule] }
+    allow(target_process).to receive(:name) { target_process_name }
+    allow(custom_target_process).to receive(:name) { custom_target_process_name }
+    allow(target_process).to receive(:process_id) { process_id }
+    allow(custom_target_process).to receive(:process_id) { custom_target_process_id }
+    allow(custom_target_process).to receive(:delete)
+    allow(schedule).to receive(:name) { schedule_name }
+    allow(schedule).to receive(:process_id) { process_id }
+    allow(target_schedule).to receive(:name) { schedule_name }
+    allow(target_schedule).to receive(:process_id) { process_id }
+    allow(target_schedule).to receive(:enable)
+    allow(target_schedule).to receive(:save)
+    allow(project).to receive(:schedules) { [schedule] }
+    allow(custom_target_schedule).to receive(:name) { 'custom schedule' }
+    allow(custom_target_schedule).to receive(:process_id) { custom_target_process_id }
+    allow(custom_target_schedule).to receive(:enable)
+    allow(custom_target_schedule).to receive(:save)
+    allow(custom_target_schedule).to receive(:delete)
   end
 
   it 'adds GOODOT_CUSTOM_PROJECT_ID to metadata' do
-    expect(project).to receive(:set_metadata)
+    expect(target_project).to receive(:set_metadata)
       .with('GOODOT_CUSTOM_PROJECT_ID', client_id)
     subject.class.call(params)
   end
 
+  context 'when delete_extra_process_schedule is true' do
+    let(:custom_params) { params.merge(delete_extra_process_schedule: false) }
+
+    it 'keeps all processes in target project' do
+      expect(target_process).not_to receive(:delete)
+      expect(custom_target_process).not_to receive(:delete)
+      GoodData::LCM2.run_action(described_class, custom_params)
+    end
+
+    it 'keeps all schedules in target project' do
```

Comment:
warning: Similar code found in synchronize_etls_in_segment_spec.rb:93, synchronize_etls_in_segment_spec.rb:99 (mass = 38) (flay)